edu-services/gradebook-service/impl/src/java/org/sakaiproject/component/gradebook/GradebookPermissionServiceImpl.java

```diff
@@ -1353,6 +1362,24 @@ public Object doInHibernate(Session session) throws HibernateException {
     	getHibernateTemplate().execute(hc);
 	}
 
+	public void clearPermissionsForUser(final String gradebookUid, final String userId) {
+		Long gradebookId = getGradebook(gradebookUid).getId();
+
+		// get the current list of permissions
+		final List<Permission> currentPermissions = getPermissionsForUser(gradebookId, userId);
+
+		HibernateCallback hc = new HibernateCallback() {
```

Comment:
Might be able to use:
`getHibernateTemplate().getSessionFactory().getCurrentSession().delete(currentPermission);`

Reply:
Thanks, @ern. I'll refactor accordingly.